api/v1/stats/sushi/PKPStatsSushiHandler.php

```diff
@@ -16,36 +16,51 @@
 
 namespace PKP\API\v1\stats\sushi;
 
+use APP\core\Application;
 use APP\facades\Repo;
 use APP\sushi\PR;
 use APP\sushi\PR_P1;
 use PKP\core\APIResponse;
 use PKP\handler\APIHandler;
 use PKP\security\authorization\ContextRequiredPolicy;
+use PKP\security\authorization\PolicySet;
+use PKP\security\authorization\RoleBasedHandlerOperationPolicy;
+use PKP\security\authorization\UserRolesRequiredPolicy;
+use PKP\security\Role;
 use PKP\sushi\CounterR5Report;
 use PKP\sushi\SushiException;
 use Slim\Http\Request as SlimHttpRequest;
 
 class PKPStatsSushiHandler extends APIHandler
 {
+    /** @var bool Whether the API is public */
+    public $isPublic = true;
+
     /**
      * Constructor
      */
     public function __construct()
     {
+        $site = Application::get()->getRequest()->getSite();
+        $context = Application::get()->getRequest()->getContext();
+        if (($site->getData('isSushiApiPublic') !== null && !$site->getData('isSushiApiPublic')) ||
+            ($context->getData('isSushiApiPublic') !== null && !$context->getData('isSushiApiPublic'))) {
+            $this->isPublic = false;
+        }
```

Comment:
Is the `$isPublic` property used anywhere other than the authorize method? If not, I'd recommend moving all of this code into the `authorize()` method and removing the property on the handler.

Reply:
It is used in constructor to define roles if it is not public... So theoretically it could also be called twice...